src/multiselect.js

```diff
@@ -0,0 +1,321 @@
+/**
+ * @license
+ * Copyright 2022 MIT
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Plugin main class.
+ */
+
+import * as Blockly from 'blockly/core';
+
+import DragSelect from '../lib/ds.min';
+
+import * as ContextMenu from './multiselect_contextmenu';
+import * as Shortcut from './multiselect_shortcut';
+import {blockSelection, inMultipleSelectionMode, setSelectionMode,
+  hasSelectedParent} from './global';
+import {MultiselectControls} from './multiselect_controls';
+
+/**
+ * Class for using multiple select blocks on workspace.
+ */
+export class Multiselect {
+  /**
+   * Initalize the class data structure.
+   * @param {!Blockly.WorkspaceSvg} workspace The workspace to sit in.
+   */
+  constructor(workspace) {
+    this.workspace_ = workspace;
+    this.hasDisableWorkspaceDrag_ = false;
+    this.justUnselectedBlock_ = null;
+    this.origHandleWsStart_ = Blockly.Gesture.prototype.handleWsStart;
+  }
+
+  /**
+   * Bind the events and replace registration.
+   * @param {{multiselectIcon: Object, useDoubleClick: boolean}} options
+   * to set.
+   */
+  init(options) {
+    this.onKeyDownWrapper_ = Blockly.browserEvents.conditionalBind(
+        this.workspace_.injectionDiv_, 'keydown', this, this.onKeyDown_);
```

Comment:
Use `getInjectionDiv()` instead of `injectionDiv_`. Most things that end in an underscore are private, and not meant to be used outside of core. Confusingly, some things that end in an underscore are protected, and can be used outside of core. That is our bad.